web_ui/packages/ui/src/loading/loading.component.tsx

```diff
@@ -1,32 +1,125 @@
 // Copyright (C) 2022-2025 Intel Corporation
 // LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
 
-import { View } from '@adobe/react-spectrum';
+import { DimensionValue, Flex, ProgressCircle, View } from '@adobe/react-spectrum';
 import { SpectrumProgressCircleProps } from '@react-types/progress';
+import { Responsive } from '@react-types/shared/src/style';
+import {
+    BoxAlignmentStyleProps,
+    StyleProps
+} from '@react-types/shared';
 
-import { LoadingIndicator } from './loading-indicator.component';
+/**
+ * Props for the Loading component.
+ *
+ * Extends Adobe Spectrum's ProgressCircle props along with styling and alignment props
+ * to provide a flexible loading indicator suitable for various use cases.
+ */
+interface LoadingProps extends SpectrumProgressCircleProps, StyleProps, BoxAlignmentStyleProps {
+    /**
+     * The display mode for the loading component.
+     * - 'inline': Renders as an inline element without overlay positioning
+     * - 'fullscreen': Renders as a full-screen overlay with default background
+     * - 'overlay': Renders as a positioned overlay with modal background
+     * @default 'fullscreen'
+     */
+    mode?: 'inline' | 'fullscreen' | 'overlay';
 
-import classes from './loading.module.scss';
+    /**
+     * Height of the spinner container (not the overlay itself).
+     * @default '100%'
+     */
+    height?: Responsive<DimensionValue>;
+
+    /**
+     * Height of the overlay container when in fullscreen or overlay mode.
+     * @default '100%'
+     */
+    overlayHeight?: Responsive<DimensionValue>;
 
-interface LoadingProps extends Partial<SpectrumProgressCircleProps> {
-    overlay?: boolean;
     id?: string;
+    backgroundColor?: string;
+    className?: string;
+    paddingTop?: Responsive<DimensionValue>;
 }
 
-export const Loading = ({ overlay = false, id, ...rest }: LoadingProps): JSX.Element => {
+/**
+ * A flexible loading component that can display a progress circle in different modes.
+ *
+ * This component consolidates various loading patterns used throughout the application:
+ * - Inline loading indicators
+ * - Full-screen loading overlays
+ * - Modal-style loading overlays
+ *
+ * The component automatically handles positioning, background colors, and accessibility
+ * attributes based on the selected mode.
+ *
+ * @example
+ * ```tsx
+ * // Simple inline loading spinner
+ * <Loading mode="inline" size="M" />
+ *
+ * // Full-screen overlay with custom background
+ * <Loading mode="overlay" backgroundColor="rgba(0,0,0,0.5)" />
+ *
+ * // Conditional loading overlay
+ * {isLoading && <Loading mode="overlay" />}
+ * ```
+ *
+ * @param props - The component props
+ * @returns A loading component with progress circle
+ */
+export const Loading = ({
+    mode = 'fullscreen',
+    size = 'L',
+    height = '100%',
+    overlayHeight = '100%',
+    id,
+    backgroundColor,
+    className,
+    alignItems = 'center',
+    justifyContent = 'center',
+    paddingTop,
```

Comment:
we have almost all possible css spacing props here. Can we make it a single prop perhaps? `customStyle` or make it even more strict and allow only passing styles through the className